src/sources.cpp

```diff
@@ -470,14 +478,18 @@ void fields::add_volume_source(const src_time &src, const volume &where, gaussia
   int np2 = (n + 2) % 3;
   // Kx = -Hy, Ky = Hx   (for d==Z)
   global_gaussianbeam_component = cH[np1];
-  add_volume_source(cE[np2], src, where, gaussianbeam_ampfunc, +1.0);
+  add_volume_source_check(cE[np2], src, where, gaussianbeam_ampfunc, +1.0, cE[np2], d,
+                          gaussianbeam_has_tm(), gaussianbeam_has_te());
```

Comment:
Rather than define functions for these, maybe just define variables?
```cxx
bool has_te = abs(beam.get_E0(2)) > 0;
bool has_tm = abs(beam.get_E0(0)) > 0 || abs(beam.get_E0(1)) > 0;
```
and then just pass `has_te, has_tm`?

Reply:
Fixed.